set gain registers

tharaka27 opened the pull request.

tharaka27 committed:
Update AD7173.cpp

tharaka27 committed:
Update AD7173.cpp

tharaka27 committed:
Update AD7173.h

tharaka27 committed:
Merge pull request #2 from tharaka27/tharaka27-patch-2

Update AD7173.h

tharaka27 committed:
Merge pull request #1 from tharaka27/tharaka27-patch-1

Update AD7173.cpp

CloudyPadmal commented on AD7173.cpp:
```diff
@@ -275,3 +275,25 @@ int AD7173Class::set_offset_config(adc7173_register_t offset, uint32_t offset_va
+	/* Address Range: 0x38 to 0x3F, Reset: 0x5XXXX0 , Name: GAIN0 to GAIN7 */
+
+	/* add the default gain value */
+	gain_value += 5312784;
```
From where is this value come from?

tharaka27 replied:
Sorry for that value. I have updated the code

CloudyPadmal commented:
Hello @tharaka27 , thank you the PR. Please squash the commits into one.

tharaka27 committed:
Update AD7173.cpp

tharaka27 committed:
Update AD7173.cpp

CloudyPadmal commented:
Looks good. As you can see this PR has 5 commits. In our projects we have a best practice to squash all commits into one if it's related to a single PR fixing one issue. You may try to squash these commits into one to get a better understanding so that it'll be easier when making new PRs. Thanks.

tharaka27 committed:
Merge branch 'master' of https://github.com/tharaka27/neurolab-firmware

tharaka27 committed:
adding neurolab class

This class is added to connect all devices to neurolab

tharaka27 committed:
added definitions

decoupled all definitions from ad7173.h and added them to definitions.h
all the definitions should define here

CloudyPadmal commented:
Hello @tharaka27 , please have a read at [FOSSASIA best practices](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia/)

Just to summarize things related to this PR, 
1. Always open an issue before working on a feature or a new thing in repo (No issue related to this PR)
2. Make only one PR addressing that issue with one commit (Seems like this PR addressing multiple issues apart from the one mentioned in the title, and 10 commits!)
3. In the PR body, mention what is implemented (No PR body - messages)
4. Once an issue is opened, make sure maintainers or mentors review that issue and give the signal to go ahead. Otherwise it'll be a waste of time if someone opens up an unwanted issue and implements something which then gets closed.

CloudyPadmal reviewed (changes requested):
Open issues related to sub features and have multiple PRs to make it easier to review.

CloudyPadmal commented:
It is highly discouraged closing PRs, especially the ones being reviewed. It would waste all the reviews and comments. Since this is your first PR, please don't do it again. Thanks.

tharaka27 commented:
> It is highly discouraged closing PRs, especially the ones being reviewed. It would waste all the reviews and comments. Since this is your first PR, please don't do it again. Thanks.

Sorry for what happened. Since I am learning by doing things get a mess sometimes. But I'll assure that it won't happen again. And also thank you for your kind reviews. They helped me learn a lot.
